starlite/kwargs.py

```diff
@@ -409,39 +418,94 @@ def _collect_params(params: Mapping[str, Any], expected: Set[ParameterDefinition
             raise ValidationException(f"Missing required parameter(s) {', '.join(missing_params)} for url {url}")
         return {p.field_name: params.get(p.field_alias, p.default_value) for p in expected}
 
+    async def resolve_dependencies(self, connection: Union["WebSocket", "Request"], kwargs: Dict[str, Any]) -> None:
+        """Resolve all dependencies into the kwargs, recursively.
+
+        Args:
+            connection: An instance of [Request][starlite.connection.Request] or [WebSocket][starlite.connection.WebSocket].
+            kwargs: Kwargs to pass to dependencies
+        """
+        for batch in self.dependency_batches:
+            await self._resolve_dependencies_concurrently(batch, connection, kwargs)
+
+    async def _resolve_dependencies_concurrently(
+        self,
+        dependencies: Collection["Dependency"],
```

Comment:
collection is too generic. This is not a collection, its a list or an iteratabe.